packages/opentelemetry-instrumentation-grpc/src/grpc-js/index.ts

```diff
@@ -70,7 +70,8 @@ export class GrpcJsInstrumentation extends InstrumentationBase {
     this._config = Object.assign({}, config);
   }
 
-  init() {
+  // use InstrumentationNodeModuleDefinition<any>[] to avoid leaking grpc types
+  protected init(): InstrumentationNodeModuleDefinition<any>[] {
```

Comment:
actually this is the change really required to avoid leaking grpc types here.

Reply:
And this is where the "generic" (interface) should be used (I may be over simplifying)
```protected init(): InstrumentationNodeModuleDefinition<T>[] {```

Reply:
`init()` is called by base class constructor (and this is the only one who should call init) and base class has no need for the types.
But generated type definitions hold full types for public and protected methods therefore this causes that types leak and we have to add the @types packages as production dependencies (see https://github.com/open-telemetry/opentelemetry-js-contrib/pull/460https://github.com/open-telemetry/opentelemetry-js-contrib/pull/460 for more details).

Moving `init()` to private would solve this also but this is not allowed in typescript (it would be in C++...).